.github/workflows/deploy-doc.yml

```
@@ -0,0 +1,66 @@
+name: Deploy Ref Doc
+
+on:
+  # Runs on tags
+  push:
+    tags:
+      - '**'
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+# Default to bash
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  # Build job
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Setup Node
+        uses: actions/setup-node@v3.3.0
+        with:
+          node-version: 17
+          check-latest: true
```

We should be able to add caching for `node_modules` with PNPM: https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-data

Not sure if the order of `actions/setup-node` and `pnpm/action-setup` is important though.
```suggestion
          check-latest: true
          cache: 'pnpm'
```